### PR TITLE
status check endpoint added for site up monitoring

### PR DESCRIPTION
**Pull request by sagargg**

### CODE REVIEW

**Review by shubham-mahajan — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -129,6 +132,29 @@ module.exports.makeApp = function () {
     next()
   })
 
+  /*
+   Route to check the status of the application by
+   calling the package_search of backend with row count as 0
+   to check whether the application is returning response or not
+ */
+  app.use('/status', async (req, res) => {
```
Comment (anuveyatsu) [inline index.js:140]: @sagargg can we move this above auth routes as it doesn't require protection (authz)?

File: index.js [hunk 2]
```diff
@@ -129,6 +132,29 @@ module.exports.makeApp = function () {
...
+    const url = resolve(
+      config.get('API_URL'),
+      `package_search?rows=0`
```
Comment (anuveyatsu) [inline index.js:143]: Not sure if `package_search` is the right CKAN API to call here.

File: index.js [hunk 3]
```diff
@@ -129,6 +132,29 @@ module.exports.makeApp = function () {
...
+    )
+    const response = await fetch(url)
+    if (response.ok) {
+      const body = await response.json()
+      if (body.success == true) {
+        res.status(200).send('Ok')
+      } else {
+        res.status(400).send('Not Ok')
```
Comment (anuveyatsu) [inline index.js:151]: HTTP code 400 is a client error but here we should respond with server error.

**Review by anuveyatsu — CHANGES_REQUESTED** [review]
@sagargg pls check comment above.

File: index.js (hunk 1 above)
Comment (anuveyatsu) [inline index.js:140]: @sagargg can we also do `app.get` instead of `app.use` so that it only accepts GET requests?

File: index.js (hunk 2 above)
Comment (anuveyatsu) [reply index.js:143]: OK, let's keep `package_search` as we want to check if db and solr services are responding.

**Review by anuveyatsu — APPROVED** [review]
LGTM

**Pull request merged**